survey_mailer/survey_mailer.go

```diff
@@ -0,0 +1,198 @@
+package survey_mailer
+
+import (
+	"errors"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/dxe/adb/config"
+	"github.com/dxe/adb/model"
+	"github.com/jmoiron/sqlx"
+	"github.com/sourcegraph/go-ses"
+)
+
+type SurveyOptions struct {
+	SurveyType     string
+	QueryDate      string
+	QueryEventType string
+	QueryEventName string
+	BodyText       string
+	BodyHtml       string
+	LinkParam      string
+}
+
+func sendMissingEmail(eventName string, attendees []string, sendingErrors []string) {
+	subject := "Missing emails and errors for survey: " + eventName
+	to := config.SurveyMissingEmail
+	bodyText := ""
+	bodyHtml := ""
+
+	if len(attendees) > 0 {
+		bodyText += "The following people did not receive a survey for this event due to not having a valid email address: "
+		bodyText += strings.Join(attendees, ", ")
+		bodyText += ". "
+		bodyHtml += "<p>The following people did not receive a survey for this event due to not having a valid email address: <br />"
+		bodyHtml += strings.Join(attendees, "<br />")
+		bodyHtml += "</p>"
+	}
+	if len(sendingErrors) > 0 {
+		bodyText += "The following addresses did not receive the email due to sending errors: "
+		bodyText += strings.Join(sendingErrors, ", ")
+		bodyText += ". "
+		bodyHtml += "<p>The following addresses did not receive the email due to sending errors <br />"
+		bodyHtml += strings.Join(sendingErrors, "<br />")
+		bodyHtml += "</p>"
+	}
+
+	if bodyText != "" {
+		sendEmail(to, subject, bodyText, bodyHtml)
+		log.Println("Sending email of missing emails and errors.")
+	}
+}
+
+func sendEmail(to string, subject string, bodyText string, bodyHtml string) error {
+	from := config.SurveyFromEmail
+	bodyHtml += `<br /><img src="https://adb.dxe.io/static/img/logo1.png" height="46" width="50">`
+	// EnvConfig uses the AWS credentials in the environment
+	// variables $AWS_ACCESS_KEY_ID nd $AWS_SECRET_KEY.
+	_, err := ses.EnvConfig.SendEmailHTML(from, to, subject, bodyText, bodyHtml)
+	if err != nil {
+		return errors.New(to)
+	}
+	return nil
+}
+
+func bulkSendEmails(event model.Event, subject string, bodyText string, bodyHtml string) {
+	var sendingErrors []string
+	for _, recipient := range event.AttendeeEmails {
+		log.Println("Sending email to:", recipient)
+		// Send email
+		err := sendEmail(recipient, subject, bodyText, bodyHtml)
+		if err != nil {
+			sendingErrors = append(sendingErrors, err.Error())
```

Comment:
Maybe change `err.Error()` to `recipient + " (" + err.Error() + ")"` (or vice versa) so that you'll know which address the error corresponds to.

Reply:
Thanks. Did this in the next commit.